apiserver/facades/controller/undertaker/undertaker.go

```diff
@@ -96,11 +104,12 @@ func (u *UndertakerAPI) RemoveModel(ctx context.Context) error {
 
 // TODO(secret): all these logic should be moved to secret service.
 func (u *UndertakerAPI) removeModelSecrets(ctx context.Context) error {
-	model, err := u.st.Model()
+	modelInfo, err := u.modelInfoService.GetModelInfo(ctx)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	secretBackendCfg, err := u.secretBackendService.GetSecretBackendConfigForAdmin(ctx, coremodel.UUID(model.UUID()))
+
+	secretBackendCfg, err := u.secretBackendService.GetSecretBackendConfigForAdmin(ctx, coremodel.UUID(modelInfo.UUID))
```

Comment:
you shouldn't need the cast to `coremodel.UUID` anymore.

Reply:
fixed.